app/models/alchemy/page/publisher.rb

```diff
@@ -19,9 +19,16 @@ def publish!(public_on:)
           version = public_version(public_on)
           DeleteElements.new(version.elements).call
 
-          # We must not use .find_each here to not mess up the order of elements
-          page.draft_version.elements.not_nested.available.each do |element|
-            Element.copy(element, page_version_id: version.id)
+          repository = page.draft_version.element_repository
+          ActiveRecord::Base.no_touching do
+            Element.acts_as_list_no_update do
+              repository.visible.not_nested.each.with_index do |element, position|
```

Comment:
Without the index start set to 1 it will be 0, what is not what acts as list is doing for `position`

```suggestion
              repository.visible.not_nested.each.with_index(1) do |element, position|
```